jamon-parent/jamon/src/main/java/org/wicketstuff/jamon/request/cycle/JamonMonitoredRequestCycle.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wicketstuff.jamon.request.cycle;
+
+import org.apache.wicket.protocol.http.WebApplication;
+import org.apache.wicket.request.component.IRequestablePage;
+import org.apache.wicket.request.cycle.RequestCycle;
+import org.apache.wicket.request.cycle.RequestCycleContext;
+import org.wicketstuff.jamon.component.JamonAdminPage;
+
+import com.jamonapi.Monitor;
+import com.jamonapi.MonitorFactory;
+
+
+/**
+ * <p>
+ * To use this class simply override the method {@link WebApplication#init} in your
+ * {@link WebApplication}. You can only use this class in combination with the
+ * {@link JamonAwareRequestCycleListener}.
+ * </p>
+ * <p>
+ * The responsibility of the {@link JamonMonitoredRequestCycle} is to add a monitor for all actions
+ * that will cause pages or parts of pages to be rendered. <br>
+ * The labels of the {@link Monitor}s come in these formats:
+ * 
+ * <ul>
+ * <li>"PageName" - When a user navigates directly to a Page, for instance the HomePage, or a
+ * bookmarked Page.</li>
+ * <li>"PageName.toNextPage -> NextPage" - When the component toNextPage on the PageName page causes
+ * the user to navigate to the "NextPage" page <b>and</b> this {@link JamonMonitoredRequestCycle}
+ * has its property {@link #includeSourceNameInMonitorLabel} set to <code>true</code>.</li>
+ * <li>"NextPage" - When the user navigates to the "NextPage" page from whatever page he was on
+ * <b>and</b> this {@link JamonMonitoredRequestCycle} has its property
+ * {@link #includeSourceNameInMonitorLabel} set to <code>false</code>.</li>
+ * </ul>
+ * 
+ * Any navigations from or to the {@link JamonAdminPage} is excluded from the Monitors.
+ * </p>
+ * 
+ * @author lars
+ * 
+ */
+public class JamonMonitoredRequestCycle extends RequestCycle
```

Comment:
IMO there is no need of custom RequestCycle.
Better extract a POJO for its members and store it in RequestCycle's metadata. Use the callback methods in the IRequestCycleListener.